retriever/lib/engine_tools.py

```diff
@@ -101,37 +104,102 @@ def reset_retriever(scope="all", ask_permission=True):
             print("can't find script {scp}".format(scp=scope))
 
 
-def json2csv(input_file, output_file=None, header_values=None, encoding=ENCODING):
-    """Convert Json file to CSV.
-
-    Function is used for only testing and can handle the file of the size.
-    """
+def json2csv(input_file,
+             output_file=None,
+             header_values=None,
+             encoding=ENCODING,
+             row_key=None):
+    """Convert Json file to CSV."""
     file_out = open_fr(input_file, encoding=encoding)
     # set output file name and write header
     if output_file is None:
         output_file = os.path.splitext(os.path.basename(input_file))[0] + ".csv"
     csv_out = open_fw(output_file, encoding=encoding)
     if os.name == 'nt':
-        outfile = csv.DictWriter(csv_out,
-                                 dialect='excel',
-                                 escapechar="\\",
-                                 lineterminator='\n',
-                                 fieldnames=header_values)
+        outfile = csv.writer(csv_out,
+                             dialect='excel',
+                             escapechar="\\",
+                             lineterminator='\n')
     else:
-        outfile = csv.DictWriter(csv_out,
-                                 dialect='excel',
-                                 escapechar="\\",
-                                 fieldnames=header_values)
-    raw_data = json.loads(file_out.read())
-    outfile.writeheader()
+        outfile = csv.writer(csv_out, dialect='excel', escapechar="\\")
 
-    for item in raw_data:
-        outfile.writerow(item)
+    raw_data = json.loads(file_out.read(), object_pairs_hook=OrderedDict)
+
+    raw_data, header_values = walker(raw_data,
+                                     row_key=row_key,
+                                     header_values=header_values,
+                                     rows=[],
+                                     normalize=False)
+
+    if isinstance(raw_data[0], dict):
+        raw_data = [list(row.values()) for row in raw_data]
```

Comment:
Why are we type casting `row.values()` to `list` here - isn't row.values() already a list?

Reply:
This case is used in case raw_data is a list of dictionaries so it will get every element as row and then retrieve it's values to convert into a list.
Example :
``` j
[
  {
    "User": "Alex",
    "id": "US1",
    "Age": "25",
    "kt": "2.0",
    "qt": "1.00"
  },
  {
    "User": "Tom",
    "id": "US2",
    "Age": "20",
    "kt": "0.0",
    "qt": "1.0"
  }
]
```

Reply:
No that's fine. I just meant why we are using `list(row.values())` instead of just using `row.values()` in the list comprehension.

Reply:
As raw_data should be a list of lists in the end